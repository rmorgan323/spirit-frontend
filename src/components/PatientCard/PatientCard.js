import React from 'react';
import './PatientCard.css';
import { Link } from 'react-router-dom';
import { connect } from 'react-redux';
import * as actions from '../../actions';

const PatientCard = (props) => {

  return (
    <div className="PatientCard">
      <p><span>Patient Id: </span>{props.absName} </p>
      <p><span>Last update: </span>{props.lastAppt}</p>
      <Link onClick={() => props.getPatientConcerns(props.id)} to={`/spirit/patients/${props.id}`}>SELECT</Link>
    </div>
  ) 
}

const mapDispatchToProps = dispatch => ({
  getPatientConcerns: (id) => {
    dispatch(actions.getPatientConcerns(id))
  }
})

export default connect(null, mapDispatchToProps)(PatientCard);
