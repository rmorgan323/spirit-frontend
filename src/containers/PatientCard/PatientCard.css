/******  GLOBAL CSS VARIABLES  ******/
/***  COLORS  ***/
/***  FONTS  ***/
* {
  font-size: 16px;
  font-family: "Lato", sans-serif;
  outline: none;
  color: #6d6e71; }

h1, h2, h3, h4, h5, h6 {
  font-family: "Josefin Sans", sans-serif;
  line-height: 1.3; }

h1 {
  font-size: 44px; }

h2 {
  font-size: 30px; }

h3 {
  font-size: 24px; }

h4 {
  font-size: 18px; }

h5 {
  font-size: 14px; }

h6 {
  font-size: 10px; }

p {
  font-size: 14px;
  line-height: 1.3;
  font-weight: 900; }

::placeholder {
  opacity: .5;
  text-align: center; }

/***  MIXINS  ***/
.PatientCard {
  display: flex;
  justify-content: space-between;
  align-items: center;
  border-bottom: 10px solid #f1f2f2;
  margin-top: 10px; }
  .PatientCard p {
    font-size: 20px; }
  .PatientCard span {
    color: #a7a9ac;
    font-size: 16px;
    margin-right: 4px; }
  .PatientCard a {
    text-decoration: none;
    color: #5cba47; }
  .PatientCard a:hover {
    color: #468c41; }
