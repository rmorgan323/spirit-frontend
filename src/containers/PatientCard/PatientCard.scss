@import '../../variables.scss';
 
 .PatientCard {
  display: flex;
  justify-content: space-between;
  align-items: center;
  border-bottom: 10px solid $grey1;
  margin-top: 10px;

  p {
    font-size: 20px;
  }

  span {
    color: $grey3;
    font-size: 16px;
    margin-right: 4px;
  }

  a {
    text-decoration: none;
    color: $green4;
  }

  a:hover {
    color: $green5;
  }
 } 