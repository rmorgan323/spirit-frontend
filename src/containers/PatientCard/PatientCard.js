import React from 'react';
import { Link } from 'react-router-dom';
import { connect } from 'react-redux';
import { PropTypes } from 'prop-types';
import * as actions from '../../actions';
import './PatientCard.css';

export const PatientCard = props => {
  const { abstractedName, lastAppt, id } = props;

  return (
    <div className="PatientCard">
      <p>
        <span>Patient Id: </span>
        {abstractedName}{' '}
      </p>

      <p>
        <span>Last update: </span>
        {lastAppt}
      </p>

      <Link
        onClick={() => props.getPatientConcerns(id)}
        to={`/spirit/patients/${id}`}
      >
        SELECT
      </Link>
    </div>
  );
};

export const mapDispatchToProps = dispatch => ({
  getPatientConcerns: id => {
    dispatch(actions.getPatientConcerns(id));
  }
});

export default connect(null, mapDispatchToProps)(PatientCard);

PatientCard.propTypes = {
  abstractedName: PropTypes.string,
  lastAppt: PropTypes.string,
  id: PropTypes.number,
  getPatientConcerns: PropTypes.func
};
