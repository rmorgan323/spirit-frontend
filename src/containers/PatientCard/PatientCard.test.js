import React from 'react';
import { PatientCard, mapDispatchToProps } from './PatientCard';
import { shallow } from 'enzyme';

describe('PatientCard tests', () => {
  let getPatientConerns;
  let abstractedName;
  let lastAppt;
  let id;
  let renderedPatientCard;

  beforeEach(() => {
    getPatientConerns = jest.fn();
    abstractedName = 'DFXAB01';
    lastAppt = '2/3/2018';
    id = 3;
    renderedPatientCard = shallow(
      <PatientCard
        getPatientConerns={getPatientConerns}
        lastAppt={lastAppt}
        abstractedName={abstractedName}
        id={id}
      />
    );
  });

  it('Should match the snapshot', () => {
    expect(renderedPatientCard).toMatchSnapshot();
  });
});

describe('mapDispatchToProps tests', () => {
  it('Should call dispatch when getPatientConcerns is called', () => {
    const mockDispatch = jest.fn();
    const result = mapDispatchToProps(mockDispatch);
    const mockParameters = 5;

    result.getPatientConcerns(mockParameters);
    expect(mockDispatch).toHaveBeenCalled();
  });
});
