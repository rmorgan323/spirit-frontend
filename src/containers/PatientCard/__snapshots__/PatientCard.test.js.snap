// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PatientCard tests Should match the snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Unknown
    abstractedName="DFXAB01"
    getPatientConerns={[Function]}
    id={3}
    lastAppt="2/3/2018"
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <p>
          <span>
            Patient Id: 
          </span>
          DFXAB01
           
        </p>,
        <p>
          <span>
            Last update: 
          </span>
          2/3/2018
        </p>,
        <Link
          onClick={[Function]}
          replace={false}
          to="/spirit/patients/3"
        >
          SELECT
        </Link>,
      ],
      "className": "PatientCard",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <span>
              Patient Id: 
            </span>,
            "DFXAB01",
            " ",
          ],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Patient Id: ",
            },
            "ref": null,
            "rendered": "Patient Id: ",
            "type": "span",
          },
          "DFXAB01",
          " ",
        ],
        "type": "p",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <span>
              Last update: 
            </span>,
            "2/3/2018",
          ],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Last update: ",
            },
            "ref": null,
            "rendered": "Last update: ",
            "type": "span",
          },
          "2/3/2018",
        ],
        "type": "p",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": "SELECT",
          "onClick": [Function],
          "replace": false,
          "to": "/spirit/patients/3",
        },
        "ref": null,
        "rendered": "SELECT",
        "type": [Function],
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <p>
            <span>
              Patient Id: 
            </span>
            DFXAB01
             
          </p>,
          <p>
            <span>
              Last update: 
            </span>
            2/3/2018
          </p>,
          <Link
            onClick={[Function]}
            replace={false}
            to="/spirit/patients/3"
          >
            SELECT
          </Link>,
        ],
        "className": "PatientCard",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <span>
                Patient Id: 
              </span>,
              "DFXAB01",
              " ",
            ],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Patient Id: ",
              },
              "ref": null,
              "rendered": "Patient Id: ",
              "type": "span",
            },
            "DFXAB01",
            " ",
          ],
          "type": "p",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <span>
                Last update: 
              </span>,
              "2/3/2018",
            ],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Last update: ",
              },
              "ref": null,
              "rendered": "Last update: ",
              "type": "span",
            },
            "2/3/2018",
          ],
          "type": "p",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": "SELECT",
            "onClick": [Function],
            "replace": false,
            "to": "/spirit/patients/3",
          },
          "ref": null,
          "rendered": "SELECT",
          "type": [Function],
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
